packages/engine/src/input/schema/ClientInputSchema.test.ts

```diff
@@ -0,0 +1,152 @@
+import assert, { strictEqual } from 'assert'
+import { LifecycleValue } from '../../common/enums/LifecycleValue'
+import { Engine } from '../../ecs/classes/Engine'
+import { GamepadAxis, TouchInputs } from '../enums/InputEnums'
+import { handleKey, handleMouseButton, handleMouseMovement, handleMouseWheel, handleTouch, handleTouchDirectionalPad, handleTouchGamepadButton, handleTouchMove, normalizeMouseCoordinates, prevTouchPosition, usingThumbstick } from './ClientInputSchema'
+
+describe('clientInputSchema', () => {
+    
+    it('check useThumpstick', () => {
+        strictEqual(usingThumbstick(), false)
+    })
+
+    describe('handleTouchMove', () => {
+        it('touchstart', () => {
+            const newClientX = 1
+            const newClientY = 1
+
+            const normalized = normalizeMouseCoordinates(newClientX, newClientY, window.innerWidth, window.innerHeight)
+            const rNorm: [number, number] = [normalized.x, normalized.y]
+
+            const touchEvent = { type: 'touchstart', 
+                            touches: [{ 
+                            force: 1, 
+                            clientX: newClientX, 
+                            clientY: newClientY, 
+                            identifier: 0, 
+                            radiusX: 0, 
+                            radiusY: 0, 
+                            pageX: 0, 
+                            pageY: 0, 
+                            rotationAngle: 60, 
+                            screenX: 0, 
+                            screenY: 0, 
+                            target: new EventTarget() }] }
+
+            handleTouchMove(touchEvent as unknown as TouchEvent)
+            strictEqual(prevTouchPosition[0], rNorm[0])
+            strictEqual(prevTouchPosition[1], rNorm[1])
+            strictEqual(Engine.inputState.get(TouchInputs.Touch1Position)?.lifecycleState, LifecycleValue.Started)
+
+            const touchEvent2 = { type: 'touchchange', 
+                            touches: [{ 
+                            force: 1, 
+                            clientX: newClientX, 
+                            clientY: newClientY, 
+                            identifier: 0, 
+                            radiusX: 0, 
+                            radiusY: 0, 
+                            pageX: 0, 
+                            pageY: 0, 
+                            rotationAngle: 60, 
+                            screenX: 0, 
+                            screenY: 0, 
+                            target: new EventTarget() }] }
+
+            handleTouchMove(touchEvent2 as unknown as TouchEvent)
+            
+            strictEqual(Engine.inputState.get(TouchInputs.Touch1Position)?.lifecycleState, LifecycleValue.Changed)
+
+            assert(Engine.inputState.size > 0)
+            Engine.inputState.clear()
```

Comment:
should move all occurances of `Engine.inputState.clear()` to beforeEach hook to avoid side effects